src/nvim/highlight_group.c

```diff
@@ -230,60 +230,100 @@ static const char *highlight_init_both[] = {
-  "default link @storageclass    StorageClass",
-  "default link @namespace       Identifier",
-  "default link @include         Include",
-  "default link @preproc         PreProc",
-  "default link @debug           Debug",
-  "default link @tag             Tag",
+
+  "default link @module  Module",
+  "default link @label   Label",
+
+  "default link @string                 String",
+  "default link @string.documentation   Comment",
+  "default link @string.regexp          SpecialChar",
+  "default link @string.escape          SpecialChar",
+  "default link @string.special         SpecialChar",
+  "default link @string.special.symbol  Constant",
+  "default link @string.special.url     Underlined",
+
+  "default link @character          Character",
+  "default link @character.special  SpecialChar",
+
+  "default link @boolean       Boolean",
+  "default link @number        Number",
+  "default link @number.float  Float",
+
+  "default link @type             Type",
+  "default link @type.definition  Typedef",
+  "default link @type.qualifier   StorageClass",
+
+  "default link @attribute  Function",
+  "default link @property   Identifier",
+
+  "default link @function          Function",
+  "default link @function.builtin  Special",
+  "default link @function.macro    Macro",
+
+  "default link @constructor   Special",
+  "default link @operator      Operator",
+
+  "default link @keyword            Keyword",
+  "default link @keyword.function   Statement",
+  "default link @keyword.operator   Operator",
+  "default link @keyword.import     Include",
+  "default link @keyword.storage    StorageClass",
+  "default link @keyword.repeat     Repeat",
+  "default link @keyword.debug      Debug",
+  "default link @keyword.exception  Exception",
+
+  "default link @keyword.conditional    Conditional",
+
+  "default link @keyword.directive         Preproc",
+  "default link @keyword.directive.define  Define",
+
+  "default link @punctuation.delimiter  Delimiter",
+  "default link @punctuation.bracket    Delimiter",
+  "default link @punctuation.special    Special",
+
+  "default link @comment   Comment",
+
+  "default link @comment.error    DiagnosticError",
+  "default link @comment.warning  DiagnosticWarn",
+  "default link @comment.hint     DiagnosticHint",
+  "default link @comment.info     DiagnosticInfo",
+  "default link @comment.todo     Todo",
+
+  "@markup.strong         gui=bold           cterm=bold",
+  "@markup.italic         gui=italic         cterm=italic",
+  "@markup.strikethrough  gui=strikethrough, cterm=strikethrough",
+  "@markup.underline      gui=underline,     cterm=underline",
+
+  "default link @markup.heading  Title",
+
+  "default link @markup.quote        Comment",
+  "default link @markup.math         Identifier",
+  "default link @markup.environment  Identifier",
+
+  "default link @markup.link        Underlined",
+  "default link @markup.link.label  Identifier",
+  "default link @markup.link.url    Underlined",
```

Comment:
As `@markup.link.url` is same as `@markup.link`, maybe it is not needed to be defined here?

Reply:
You are right; I think I wanted to be explicit, but that may not be needed here.

Having it _does_ make a difference: `link.url` will stay underlined if someone redefines `link`. That may be desirable in this(!) case?

Reply:
I don't, but that doesn't mean that it can happen (maybe people don't want _internal_ links to be underlined and reserve this for actual URLs?)

As it doesn't make any difference out of the box, I'm fine with either (with a small preference for keeping it this way, as it doesn't cost us anything).